\chapter{Introduction}

\textit{There's an infinity of things that have never been done before, and most of those things are not worth doing.} -- Jeremy Fox, \textit{Dynamic Ecology (2016)}~\cite{fox_how_2016}

\section*{Summary}

This chapter presents... (approx. 1 page)


\selectlanguage{french}
\section*{Résumé}

Dans cette section... (env. 1 page)
\selectlanguage{english}

\section{Context}
\label{sec:context}

\subsection{Self-replication is a resource allocation problem}

At every moment of our lives, we are surrounded by billions of microscopic organisms.
Sometimes beneficial, sometimes harmful, these small factories collect and use energy from their environment, in the form of organic matter, highly reactive chemicals, or light [citation trophic].
While most cells in our body are kept in a stable and friendly micro-environment (temperature, oxygenation, nutrient abundance, ...), microorganisms have evolved to constantly adapt their physiology to a strongly variable environment.
Microbes are found almost everywhere, sometimes in the strangest places that were initially thought unsuitable for life [citation deep water bacteria, citation bacteria in space, citation review bacterial diversity].
Closer to us, the model organism \textit{Escherichia coli}, a bacteria first isolated from our intestinal track, is capable of growing on dozens of different sugars through changes in the expression of thousands of genes\cite{zimmer_microcosm:_2009}.

Polymers are the most abundant organic molecules that constitute a living cell.
They consist of many repeated similar sub-units (monomers).
There are about 20 different amino acids in the proteins, and 5 different nucleobases in the (desoxy-)ribonucleotids that constitute RNA and DNA (the famous A, T, C, G letters and the extra U only found in RNA).
The information about their sequence is stored in the form of genes on the DNA.
The process by which this information is used in the synthesis of a functional final product (protein or RNA) is called gene expression.
Each gene contains a target sequence that allows the gene expression machinery to recognize it and start the synthesis of its product.
This sequence is called a promoter.
If every promoter were the same, all genes would be expressed at the same level.
That of course does not happen.
In practice, some promoters are stronger than others, and attracts more of the available transcriptional machinery.
Furthermore, there exist gene products that bind the promoter region of genes (transcription factors), stimulating or inhibiting the promoter affinity, hence the gene expression.
This simple mechanism, among others, allows the regulation of cell composition.

The reorganization of gene expression controls the abundance of enzymes and ribozymes that catalyzes the biochemical reactions allowing the cell to perform different functions.
For instance, assimilating a given nutrient involves the synthesis of the corresponding uptake protein and of the metabolic enzymes converting the nutrients into reductive power, nucleotids and amino acids~\cite{schaechter_microbe_2006}.
These precursors are then used to produce new macromolecules, closing the loop of self-replication.
In order to optimize the overall process, the cell must adjust the abundance of every enzyme involved in this network of biochemical reactions.
By doing so, the cell actually resolves a resource allocation problem, where a common wealth (precursors) has to be distributed over thousands of potential beneficiaries (proteins and other macromolecules).

\subsection{Optimizing biomass production yields a competitive advantage}

What is the optimal allocation of precursors for a living cell?
Under Darwin's theory, the optimization criteria of living systems are called fitness factors, and measure to which extend an organism fits into a given environment.
Fitness factors take into account the ability to survive and reproduce based on the adaptation to the environment and the competition with other organisms.
Whether fitness factors apply at the species, the population, the organism, or even the gene level is still highly debated~\cite{dawkins_selfish_1976}.
This makes their clear identification highly difficult, even for organisms living an apparently simple life.

For this reason, there is no consensus about what constitute the main fitness factors of microorganisms.
Competition for common resources seems to favor the maximization of offspring, in the sense that a replicator with more descendants will outnumber its competitors in the long term~\cite{dawkins_selfish_1976}.
It is unclear, however, how this principle translates into the physiological behavior of microorganisms.
For instance, metabolic networks seems to either maximize ATP or biomass yield depending on the environmental condition, a result obtained by looking at the predicting power of genome-scale metabolic networks~\cite{schuetz_systematic_2007}.
But, yield is not an universal fitness factor, because in fluctuating or limited environments, evolution seems to favor bacteria with a high rate of nutrient assimilation, instead of a high yield of transformation even though it would favor the net biomass production of the population~\cite{frank_tradeoff_2010,maclean_tragedy_2008}, a result that has also been confirmed in yeast~\cite{schuster_maximization_2008}.
Overall, it seems that cell optimality drifts in a multidimensional space and involves making several trade-offs that strongly depend on the environmental context~\cite{schuetz_multidimensional_2012}.

Nevertheless, numerous studies have shown that considering growth rate maximization confers a good predictive power of microorganism physiology.
Uptake rates of \textit{Escherichia coli} growing on acetate or succinate are correctly predicted by assuming that its metabolic network operates in a mode that maximizes growth~\cite{edwards_silico_2001}.
This result is not general and can initially fail for other carbon sources or strain~ \cite{ibarra_escherichia_2002}.
But even in those cases, the metabolic networks evolve rapidly towards growth maximization if a constant nutrient-providing environment is maintained~\cite{ibarra_escherichia_2002}.
The mutants obtained present modications in the regulation of the synthesis of metabolic enzymes~\cite{lewis_omic_2010}, indicating that not solely the efficiency of the enzymes is affected, but also as a consequence the cellular composition.
Overall, when focusing on growth, growth rate maximization sounds like a rational choice to understand the growth strategies of microorganisms~\cite{molenaar_shifts_2009}.

\subsection{Growth laws are universal strategies employed by microorganisms}

One of the conclusions of the preceding section is that growth rate maximization is a rather general principle of microbial life.
Would microbial physiology be determined by this criteria, we could expect to find universal regularities in the way physiology adapts to the environment.
Are there fundamental laws that control the physiological parameters of microorganisms, or self-replication as a whole?

The most emblematic effort to formalize microorganism growth by coarse-grained modelling is the discovery of Monod's law more than sixty years ago~\cite{monod_growth_1949}.
This empirical relationship states that the growth rate $\mu$ [\texttt{div.h\textsuperscript{-1}}] of \textit{Escherichia coli} is a hyperbolic function of the concentration of the limiting nutrient $C$ [\texttt{mol.L\textsuperscript{-1}}], such that
\begin{equation}
\label{eq:monod_law}
\mu = \mu_K \frac{C}{K_C + C}
\end{equation}
with $\mu_K$ and $K_C$ two constants depending on the quality of the nutrient (Fig~\ref{fig:monod_law}).
It is remarkable that the growth rate, a parameter that depends on the use and production of thousands of proteins, can be so easily predicted by a simple relation.
What is even more remarkable is that this relation is rather universal.
Despite some adjustments (Droop model being a well-known example~\cite{droop_thoughts_1973}), the relation holds for most microorganisms: the growth rate increases with the abundance of the limiting nutrient, up to a point beyond which growth cannot be any faster~\cite{koch_why_1988}.

\begin{figure}[!h]
\centering
\includegraphics[height=3cm]{./Fig/Chapter1/monod_law.eps}
\caption{Monod law, reproduced from~\cite{monod_growth_1949}.
Monod has been a pioneer in the formalization of microbial growth into fundamental coarse-grained relationships.
This figure shows the growth rate (in number of divisions per hour) of \textit{Escherichia coli} in a synthetic minimal medium at 37$^{\circ}$C containing different concentrations of glucose (a common carbon source).
As the concentration of glucose increases, the growth rate of \textit{E. coli} does so by following a hyperbolic relationship described in Eq~\ref{eq:monod_law}.
Solid line is drawn with $\mu_K = 1.35$ div.h\textsuperscript{-1}, and $K_C = 0.22 \cdot 10^{-4}$ mol.L\textsuperscript{-1}.
}
\label{fig:monod_law}
\end{figure}

The production of every component of the cell is under complex regulations.
But they are all linked to the growth rate, since the production has to compensate for the dilution caused by growth~\cite{monod_growth_1949}.
Many physiological parameters (like mass of DNA, RNA and proteins) are then functions of the growth rate alone, regardless of the environmental conditions~\cite{schaechter_dependency_1958,bremer_modulation_1996}.
These functions were carefully measured~\cite{bremer_modulation_1996} and are still used today in the quantitative understanding of growth control in microorganisms~\cite{ehrenberg_mediumdependent_2012}.
Recently, the topic was brought to the fore by the work of Scott \textit{et al}, whom redefined and revived interest in the bacterial growth laws~\cite{scott_bacterial_2011}.
They focused on the RNA/protein ratio, a physiological parameter that was long known to vary linearly with the growth rate in microorganisms (Fig~\ref{fig:scott_rnaprot}).
By analyzing this ratio under different environmental perturbations of the protein synthesis machinery, they built a coarse-grained model describing proteome resource allocation~\cite{scott_emergence_2014}.
It has allowed to show that, whatever the details of the molecular implementation (differing between organisms), universal growth laws emerges from the underlying principles of robustness and optimization imposed by natural selection~\cite{scott_emergence_2014}.  % detail this

\begin{figure}[!h]
\centering
\includegraphics[height=6cm]{./Fig/Chapter1/scott_rnaprot.eps}
\caption{The growth law of ribosomal abundance (figure reproduced from Fig~S1 in~\cite{scott_interdependence_2010}).
For different carbon sources more or less rich, the RNA/protein ratio is linearly correlated with the growth rate, a relation that holds for many species of microorganisms.
This ratio is correlated with the fraction of ribosome-affiliated proteins, and therefore with the relative abundance of ribosomes in the cell.
}
\label{fig:scott_rnaprot}
\end{figure}

Coarse-grained models have also been helpful to understand the process by which microorganisms switch to inefficient metabolic pathways when growing at high nutrient availability.
Since nutrients are not used at their full potential given the environmental conditions, this widespread growth strategy apparently leads to energy waste, ~\cite{molenaar_shifts_2009} (Fig~\ref{fig:molenaar_overflow}).
Originally called the Crabtree effect when first observed in \textit{Saccharomyces cerevisiae}~\cite{dijken_kinetics_1993} and overflow metabolism when later found in \textit{E. coli}~\cite{vemuri_overflow_2006}, it has also been called the Warburg effect in many types of tumor cells~\cite{mckeehan_glycolysis_1982,hsu_cancer_2008}.
Molenaar \textit{et al} showed that this phenomenon simply results from the underlying principle of growth rate maximization, in particular the trade-off between the cost of enzyme synthesis and the pathway efficiency~\cite{molenaar_shifts_2009}.
Like traders, bacteria distribute their resources between different beneficiaries, and tend to do so in a way that maximizes returns on investments.

\begin{figure}[!h]
\centering
\includegraphics[height=5cm]{./Fig/Chapter1/molenaar_overflow.eps}
\caption{The strategy of overflow metabolism, reproduced from~\cite{molenaar_shifts_2009}.
When increasing the concentration of substrate, the metabolism of microorganisms switches toward the use of inefficient pathways.
This paradoxical behavior does in fact maximize the growth rate if we take into account the costs and benefits of enzyme synthesis, in particular the fact that pathways for inefficient catabolism are shorter and therefore cheaper to produce.
(A) Simple view of microbial metabolism, with the two competitive efficient and inefficient pathways.
(B) The switching from efficient to inefficient catabolism when the concentration of nutrients, and hence the growth rate, increases.
}
\label{fig:molenaar_overflow}
\end{figure}

\subsection{Static versus dynamical perspective on growth}

Growth rate maximization does a pretty good job at explaining the universal growth strategies described above.
By maximizing offspring, it seems to comply perfectly with the principles of natural selection cited earlier.
However, all the studies cited above apply this growth rate maximization in a very specific growth scenario, one that in fact does not correspond to the environmental conditions in which microorganisms have been selected, as we will argue below.

Most microbial studies have been done at steady state, where all intensive properties of the cell are time-invariant~\cite{schaechter_microbe_2006,fishov_microbial_1995}.
It means that the properties independent of the cell volume or mass (temperature, concentrations, ...) are constant over time, even though the cell is still growing.
It requires that the components of the cell "increases by the same factor over a time interval", which have motivated the creation of the term balanced growth~\cite{campbell_synchronization_1957}.
Experimentally, this growth scenario has been used as a standard because it improves reproducibility, in that the time-point of sampling becomes irrelevant~\cite{schaechter_microbe_2006}.
It can be easily achieved laboratory conditions either in continuous culture, where the substrate is continually supplied, or in batch conditions if the substrate is in high excess (Fig~\ref{fig:growth_curve}).
This approach has been beneficial to mathematical modeling because considering the system at steady state reduces the complexity of the underlying dynamical systems, allowing to build and analyze huge genome-scale models that encapsulate the enzymatic diversity of living cells [citation FBA].

\begin{figure}[!h]
\centering
\includegraphics[height=5cm]{./Fig/Chapter1/growth_curve.eps}
\caption{The different phases of a typical growth curve are visualized by the accumulation of biomass (black thick line) and the nutrient abundance in the medium (green dashed line).
(A) The lag phase, a variable period of time during which the organism adapt to the new medium.
It is hardly reproducible and strongly depends on the recent history of the culture.
(B) The steady-state (or balanced-growth) phase which is characterized by an exponential growth of biomass.
It presents constant and reproducible characteristics which has made it a standard for microbial growth studies~\cite{schaechter_microbe_2006}.
This phase can be extended for thousands of generations by the constant addition of fresh medium (continuous culture).
As represented here, nutrients are quickly depleted in batch conditions, which does not allow to maintain steady-state growth for a long period of time.
(C) The stationary phase, after the depletion of the limited nutrient in the medium.
In environmental conditions, microorganisms usually encounter poor media and spend most of their time in stationary phase.
Some species which have evolved long-term resistance mechanisms like sporulation can survive harsh conditions when in stationary phase.
}
\label{fig:growth_curve}
\end{figure}

Although balanced growth is convenient from an experimental and theoretical point of view, it is widely admitted that microorganisms rarely encounter this state in nature since it requires a stable environment over a long period of time~\cite{schaechter_microbe_2006}.
Microorganisms live in strongly dynamic communities, in medium where the limited nutrient is quickly depleted by the competitors when available\footnote{an exception being bacteria that live a parasitic or symbiotic life in bigger cells [citation Micoplasma genitallium, Giant tube worm]}.
With that in mind, a microorganism aptness to optimize its steady-state growth rate should not play any role for its persistence.
Why would in that case microorganisms be optimal for steady-state growth?
Are we missing something by studying systems that were selected to cope with environmental variations (regulations of cell composition) in stable, unchanging conditions that probably confuse them?

\section{Problem statement}

With their beautiful simplicity, growth laws represent an exceptional point of view on microbial life strategies.
But being established at balanced growth might have drawbacks since it is a growth state totally irrelevant to the life-trait history of microorganisms.
How would growth laws be affected in a dynamical context?
Can we still find universal regularities during growth states more relevant to the natural habitat of microorganisms?

Since we lose the inherent advantages of working at steady state, we must confine our problem to keep it tractable.
We then focus on the fundamentals behind resource allocation (the proteome composition growth law), especially the regulation of the abundance of gene expression machinery during the transition between two different growth media.

On the theoretical side, we assess if there exist a universal optimal way to allocate cell resources during growth transitions.
On the experimental side, we monitor the abundance of key macromolecular components in order to get an insight that could be compared with the theoretical predictions.

\section{Related work}

\subsection{Modeling growth of microorganisms}

My recent experience as a teacher taught me that, as of today, a lot of students still choose to study biology in the hope to find a math-free job in science.
They fall from the clouds when you tell them modeling is an important part of a biologist's life.
This misconception usually comes from a bad understanding of what is modeling.
In its general sense, a model depicts and simplifies reality.
Maps, sketches, or pictures satisfy this definition, as do graphs, sets of equations, or even the DNA sequences I store in a text-file on my computer.
Overall, depicting and simplifying reality makes you a modeler, not so much drawing equations on a board.
But this does not change the fact that mathematics are incredible tools for these purposes~\cite{servedio_not_2014,mcgill_calm_2013}.

What makes mathematical modeling so useful in microbial growth studies?
Microbial physiology comes out from the interplay of thousands of chemical reactions that are not necessarily relevant in any given situation.
They usually occur on a short time scale as compared with the organism life span.
They nevertheless have been (and still are) selected by evolutionary processes that occur on long time scales.
These physical limitations strongly impede a full empirical study of growth processes, making sometimes hard to test the most simple verbal hypothesis.

Since they are designed to abstract away complexity, mathematical models are great tools to deal with such limitations.
The overwhelming number of variables and time scales can be dealt with by choosing the correct framework.
When working on a model, growth is abstracted into a world of clearly stated rules, where logical reasoning can be applied~\cite{servedio_not_2014}.
Predictions can be made, "unpacked" into our real world, and confronted with experimental testing.
In that sense, modeling is at the core of the scientific method~\cite{mcgill_calm_2013}.
This could explain why mathematical modeling has been extensively applied to microbial growth studies, creating a diversity a modeling framework.

For an increasing number of microorganisms, we are now able to draw quasi-exhaustive maps of their inner metabolic reactions.
This has been made possible by the advances on genome reconstruction, which are easier and cheaper to accomplish every passing year.
But making sense of these data proved to be even more challenging.
Which reactions are important, and in which conditions?
How will a metabolic network react to perturbations?
Can we predict its evolution in a given environment?
When dealing with genome-scale metabolic networks, one is challenged by the number of metabolites and possible reactions (e.g. the most recent genome-scale reconstruction of \textit{E. coli} metabolism contains "1366 genes, 2251 metabolic reactions, and 1136 unique metabolites"~\cite{orth_comprehensive_2011}).
More importantly, while progress in genomic made possible such reconstructions, measuring precise \textit{in-vivo} kinetic parameters for all these reactions still represents a technological bottleneck~[citation?].
Nevertheless, a modeling framework is specifically designed to deal with such limitations: constraint-based modeling.
This framework abstracts away the unknown kinetics and represents the metabolic maps as a flux map on which physico-chemical constraints can be applied, e.g. steady-state, compartmentalization, mass conservation, molecular crowding, and thermodynamic directionality~\cite{ebrahim_cobrapy_2013}.
Candidates for the flux distribution are obtained by optimizing the system towards an objective function, which usually represents growth rate for microorganisms.
One can find a lot of derivatives of this framework, but its the Flux Balance Analysis that renewed interest into it [citation review ?].
\textbf{[Need some key results that were obtained, or this paragraph makes no sense].}
Note that such a level of abstraction necessarily has its drawbacks, e.g. the definition of the objective function, the difficulty to work in dynamic or implement information on the metabolic regulations [citation?].

The ever-growing available data on metabolic networks can also be used to make large kinetic models.
For the reasons we stated above, we expect these models to be inherently limited in scope.
It means that the modeler has to be extremely careful since he or she injects its knowledge about what might be important in the system, for the question he or she tries to solve.
They are however more flexible, in the sense that one can easily make interplay different level of regulations.
\textit{[Present the work of Klumpp, Kremling, Sauer, whole model of genitallium, etc]
}\textit{Especially the work of Klumpp, Kremling, Sauer, etc, also the whole model of genitallium.
Biggest problem is the lack of parameter values, and some huge trouble with identifiability.
They have medium to large scope (increasing), highly detailed granularity and deal with the mathematical complexity via the computational power via the use of huge quantity of data.}

Another approach has shown successful in understanding some aspects of growth behaviors of microorganisms.
By applying a high level of abstraction on how the cell functions, it becomes easy to compare data about different species and to uncover general evolutionary principles that are independent of the specific implementation in each organism.
Best examples of this approach are the empirical growth laws of ribosomal abundance~\cite{scott_bacterial_2011,scott_interdependence_2010,scott_emergence_2014}, or the proof-of-concept explanation of overflow metabolism~\cite{molenaar_shifts_2009} that were described in section~\ref{sec:context} of this introduction.
It is also the preferred approach when one wants to broaden its access to mathematical tools~\cite{vandenberg_optimal_1998}, most of them being unhelpful with the high dimensionality of biological models.
As we will describe in section~\ref{sec:approach}, this represents the modeling strategy that was used in this thesis.

\subsection{Measuring growth of microorganisms}

Models have to be tested with data.
Measuring growth = measuring biomass accumulation, over time, and get relevant information on the cell inner state.
We have a lot of data on this.

Population measurement is easy to implement, and allow to measure metabolites.
But small unsynchronized effects can be hidden since we mostly work with the mean.
Despite most of the knowledge we have on microorganism growth are from population measurements, like the RNA/DNA/etc ratios, the growth laws, etc.
Extensive work of Bremmer and Dennis.
Big discoveries of the last decades that rely on population measurements ? (cite 2 or 3)

Single-cell measurements lay on technologies like microscopy, fluorescence, FACS...
More interesting since we can use fluorescence to not just look at the cell but also measure interesting parameters (like gene expression) at the single-cell level.
Information on the localisation of DNA, ribosomes [Bakshi].
Variability in physiology, persistors, antibiotics resistance, etc [citations]

Both approaches can be performed either in batch or continuous conditions.
Batch conditions is the most straight forward in lab: you put sugar in a flask, they grow until nutrient is depleted.
Implemented in flash or petri-microscopy.
Resist to contamination.
It is close to natural conditions, and industrial conditions.
Has even been used to study long-term evolution of E. coli.
But the system is poorly control, mostly because the history of the cells play a huge role in the waking process.

Continuous culture consists into constantly providing nutrients to a culture, and maintaining it in a steady-state for a long time.
Implemented in bioreactor [citation] or microfluidic devices [mother machine citation].
Sensible to contaminations so hard to maintained (that's why industry ignore it so far).
Does not correspond to natural conditions but the system is fully controlled: perturbations can be applied and a lot of parameters can be monitored.
Possibility to let the cell long-enough in steady state to "forget" strain history.
Most suitable if one need to have full control on its system.

Finally, most of the work cited above are based on steady-state studies.
But there are also work in dynamic.
Levy showed that... in yeast.
Alon in Coli.
After decades working at steady-state, Bremmer and Dennis argued we should measure things during transitions.
de Jong / Gerosa showed that the gene expression machinery has its own dynamic that should be taken into account when measuring gene expression.
The evolutionary relevance of growth transitions has been emphasized in numerous others studies [citations lag phase, last generation before stationary, ...].

\section{Approach}
\label{sec:approach}

We first tackle the problem of creating a mathematical framework for studying environmental transitions.
\textit{Building upon the work cited above, we create a simple self-replicator model that allocate resources between two sections.}

The goal is to identify what would be an ideal transition.
\textit{We all have an intuitive idea of what should be a good transition: it should be fast, robust, etc... But it is unclear how this relates to the fitness factor that shaped the regulatory mechanisms of microorganisms.}

We describe dynamical optimality by using optimal control theory.
\textit{Brieve explanation about optimal control theory and how it is used in economic or engineering to optimize or understant dynamical processes.}

Then, after we know what to look for, we try to actually measure how resource allocation occurs in a model organism: \textit{Escherichia coli}.
\textit{We justify the use of coli and introduce the gene expression fluorescent reporter.
We made a strain with fluorescent ribosomes, and studied our experimental framework in different transitions, at the population and the single cell level.
We justify the use of both by the fact that population is easy for "screening", and single cell allow for a better resolution and take care of the "desynchronization" biais that could occur.}

\section{Organisation of the dissertation}

(announcing the plan of the manuscript, which actually follow exactly the approach part, without the lenghty justifications)